src/components/button/button.component.tsx

```diff
@@ -101,21 +193,29 @@ const Button = ({
     );
   }
 
-  const [internalRef, setInternalRef] = useState(null);
```

Comment:
As far as I remember internalRef has to be set by using useState instead of useRef in this case because Tippy that is used in Tooltip is initially rendered with an empty ref. And to reinitialize itself when a ref is changed it has to receive a new `target` prop. When you use `useRef` instead of `useState` here it receives the same object but with a mutated `current` prop meaning it won't rerender and might cause accessibility issues - because that was the reason why I added this change 2 months ago

Reply:
Hmm interesting. I'll talk to you on Slack about this and see if we can replicate the issue you're talking about. Unfortunately bits of the change you added added invalid code in TypeScript - you can't set `forwardRef.current = reference` because `current` is a readonly property of a React reference. 

If we can replicate the issue you've mentioned then let's find another way to solve the problem.